storagemarket/impl/providerstates/provider_states.go

```diff
@@ -101,35 +103,42 @@ func ValidateDealProposal(ctx fsm.Context, environment ProviderDealEnvironment,
 		return ctx.Trigger(storagemarket.ProviderEventDealRejected, xerrors.New("clientMarketBalance.Available too small"))
 	}
 
-	// TODO: Send intent to accept
-	return ctx.Trigger(storagemarket.ProviderEventDealAccepted)
-}
+	// Send intent to accept
+	err = environment.SendSignedResponse(ctx.Context(), &network.Response{
+		State:    storagemarket.StorageDealWaitingForData,
+		Proposal: deal.ProposalCid,
+	})
+
+	if err != nil {
+		return ctx.Trigger(storagemarket.ProviderEventSendResponseFailed, err)
+	}
 
-// TransferData initiates a data transfer or places the deal in a waiting state if it is a
-// manual deal
-func TransferData(ctx fsm.Context, environment ProviderDealEnvironment, deal storagemarket.MinerDeal) error {
 	if deal.Ref.TransferType == storagemarket.TTManual {
 		log.Info("deal entering manual transfer state")
 		return ctx.Trigger(storagemarket.ProviderEventWaitingForManualData)
 	}
 
-	log.Infof("fetching data for a deal %s", deal.ProposalCid)
-
-	// initiate a pull data transfer. This will complete asynchronously and the
-	// completion of the data transfer will trigger a change in deal state
-	// (see onDataTransferEvent)
-	err := environment.StartDataTransfer(ctx.Context(),
-		deal.Client,
-		&requestvalidation.StorageDataTransferVoucher{Proposal: deal.ProposalCid},
-		deal.Ref.Root,
-		shared.AllSelector(),
-	)
+	return ctx.Trigger(storagemarket.ProviderEventDataRequested)
+}
 
-	if err != nil {
-		return ctx.Trigger(storagemarket.ProviderEventDataTransferFailed, xerrors.Errorf("failed to open pull data channel: %w", err))
+func WaitingForData(ctx fsm.Context, environment ProviderDealEnvironment, deal storagemarket.MinerDeal) error {
+	if deal.Ref.TransferType == storagemarket.TTManual {
+		return nil
 	}
 
-	return ctx.Trigger(storagemarket.ProviderEventDataTransferInitiated)
+	environment.OnDataTransferEvent(ctx.Context(), datatransfer.Open, func(event datatransfer.Event) error {
+		return ctx.Trigger(storagemarket.ProviderEventDataTransferInitiated)
+	})
+
+	return nil
+}
+
+func TransferringData(ctx fsm.Context, environment ProviderDealEnvironment, deal storagemarket.MinerDeal) error {
```

Comment:
This handler should be unneccesary if you have a general data transfer subscriber